components-java-sdk/src/main/java/io/dapr/components/aspects/AdvertisesFeatures.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.dapr.components.aspects;
+
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+
+public interface AdvertisesFeatures {
+  Mono<List<String>> getFeatures();
```

Comment:
Silly haven't-actively-developed-Java-in-20-years question: what's the prevalence of the `Mono` framework compared to the (any) native Java asynchronous techniques?  (I love the Reactive Extensions for .NET, for example, but exposing them from a public API sometimes gave people heartburn vs. the "standard" `Task` and `event` approach.)

Reply:
To be honest, I really don't know what is the most common async solution for Java. I don't know if by Java17 there is SDK-native solution and, even if there is, if it is the _de facto_ standard.

Reactor (Mono/Flux) is the async solution used by Dapr Java SDK. By using it we are keeping consistency. So, in a certain way, the decision for using Reactor is grandfathered.